### PR TITLE
[REJECTED ] Test for Queue iterator optimization

### PR DESCRIPTION
**Pull request by som-snytt**
The junit QueueTest is not equipped to instrument anything,
so we use a local List that will complain loudly, like
certain old gearboxes, when thrown into reverse.

### CODE REVIEW

File: test/files/run/t11396.scala [hunk 1]
```diff
@@ -0,0 +1,664 @@
...
+  val six   = three.enqueueAll(4 to 6)
+  val it    = six.iterator
+  val _     = six.enqueue(7)
+  assertEquals(1, it.next())
+
+  def assertEquals(x: Int, y: Int) = assert(x == y)
+}
+
+/* With .javaopts, test is forked with test classes ahead of standard library on the class path.
+ * This is a copy of List at the time of writing, with reverse rigged to blow, like those movies
```
Comment (NthPortal) [inline test/files/run/t11396.scala:14]: I feel like there has to be a way to do this which doesn't involve cloning the class (which is a pain to keep up-to-date). Perhaps using reflection to replace `Queue#in` with a `Proxy` and `InvocationHandler`?

Comment (som-snytt) [reply test/files/run/t11396.scala:14]: The library is locked down until 3.2, so I went for quick and dirty. I considered minimizing the clone to what's required for the test, which is doable.

**Comment by szeiger** [comment]
I don't think it's worth the complexity. The test still cannot ensure that there are no unnecessary traversals, it only tests that the specific code path which caused this before #8449 is no longer hit. Requiring a modified copy of List just looks like too much of a maintenance burden for such an assertion.

**Comment by som-snytt** [comment]
Agree with you both. I took another look but don't remember if I came up with a better test. I'll close this.

**Pull request closed without merging**